lib/app-routes/common/auth/keys.js

```diff
@@ -15,11 +17,22 @@ exports.register = function(server, options, next) {
     const authUtils = server.plugins.utilities.auth;
     const invalidCookie = casCookieUtils.invalidate();
 
+    const LoginRecord = server.plugins.bookshelf.model('LoginRecord');
+
     server.state(config.get('casCookieName'), {
         path: '/',
         domain: config.get('cookieDomain')
     });
 
+    /**
+     * log login records in both success / failure
+     */
+    const saveRecordObj = ()=> {
+
+        LoginRecord.forge(recordObj)
```

Comment:
You should return the promise that is returned by `LoginRecord.save`. If you don't return the promise, then future `then` statements will not wait for this to finish before proceeding...